tensorflow/core/kernels/sparse_tensor_dense_matmul_op_gpu.cu.cc

```diff
@@ -41,6 +42,35 @@ struct OutOfBoundsValue<std::complex<T>> {
   }
 };
 
+template <typename Tsrc, typename Tdst>
+__global__ void DownCast(
+    const int size, const Tsrc* __restrict__  src, Tdst* __restrict__ dst) {
+
+  GPU_1D_KERNEL_LOOP(index, size) {
+    dst[index] = static_cast<Tdst>(src[index]);
+  }
+}
+
+template<>
+__global__ void DownCast(
+    const int size, const complex128* __restrict__  src,
+    complex64* __restrict__ dst) {
+
+  GPU_1D_KERNEL_LOOP(index, size) {
+    dst[index] = complex64(src[index].real(), (src[index].imag()));
+  }
+}
+
+template <typename Tout, typename Tin>
+__device__ void GpuRecastAtomicAdd(Tout* out_location, Tin val) {
```

Comment:
I think the name of this can be simplified to `GpuCastAtomicAdd`. "Recast" just means "cast", and we don't need to create a new term.

Reply:
Note to self: resolved.